crates/events/src/lib.rs

```diff
@@ -0,0 +1,192 @@
+/*!
+ */
+#![deny(missing_docs, missing_debug_implementations)]
+
+use std::borrow::Cow;
+use wasm_bindgen::closure::Closure;
+use wasm_bindgen::{JsCast, UnwrapThrowExt};
+use web_sys::{AddEventListenerOptions, Event, EventTarget};
+
+///
+#[derive(Debug, Clone, Copy)]
+pub struct EventListenerOptions {
+    ///
+    pub capture: bool,
+
+    ///
+    pub once: bool,
+
+    ///
+    pub passive: bool,
+}
+
+impl EventListenerOptions {
+    ///
+    #[inline]
+    pub fn to_js(&self) -> AddEventListenerOptions {
+        let mut options = AddEventListenerOptions::new();
+
+        options.capture(self.capture);
+        options.once(self.once);
+        options.passive(self.passive);
+
+        options
+    }
+}
+
+// This defaults passive to true to avoid performance issues in browsers:
+// https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#Improving_scrolling_performance_with_passive_listeners
+thread_local! {
```

Comment:
Aside: We might want to use something other than `thread_local!` here; IIRC it has an unfortunate code size footprint over unsafe static refcells (assuming single-threaded wasm). This can be addressed in follow up PRs (and actually measured). We might want to make a `wasm_thread_local!` thing that wraps up the unsafety.

Reply:
I don't think we should assume single-threaded wasm, since multi-threaded wasm is just around the corner.

In any case, I agree we can investigate that later, since it's just an internal optimization.